File: monsoon3.0/src/main/java/wtf/monsoon/impl/event/EventRender2D.java
package wtf.monsoon.impl.event;

import lombok.Data;
import lombok.NonNull;
import net.minecraft.client.gui.ScaledResolution;
import wtf.monsoon.api.event.Event;

@Data
public class EventRender2D extends Event {

    @NonNull
    private ScaledResolution sr;

    @NonNull
    private float partialTicks;

    @NonNull
    private float width, height;
}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/event/EventRender2D.kt
package wtf.monsoon.client.event

import me.bush.eventbuskotlin.Event
import net.minecraft.client.gui.ScaledResolution
import spritz.api.annotations.Identifier
import wtf.monsoon.Wrapper
import wtf.monsoon.client.util.ui.NVGWrapper

data class EventRender2D(@Identifier("scaled_resolution") val scaledResolution: ScaledResolution, val displayWidth: Int, val displayHeight: Int) : Event() {

    override val cancellable: Boolean = true
    val nvg: NVGWrapper = Wrapper.monsoon.nvg

}